src/driver/sqlserver/SqlServerDriver.ts

```diff
@@ -166,6 +166,7 @@ export class SqlServerDriver implements Driver {
         version: "int",
         treeLevel: "int",
         migrationId: "int",
+        migrationHash: "char",
```

Comment:
is this column being created as nullable? (because I guess it will break exist tables in the case if its not)

Reply:
It is being create as a nullable first, then hashes of all available existing migrations are calculated, and then it is being set to `not null`.

I just added a change which will only set it to `not null` when all hashes are set.

Reply:
Why column type changes depending on the driver? It's char in some, text in sqlite, int in mongo.

Reply:
IIRC, I do not think that the types in `MongoDriver` are actually being used. There is a comment:

> Mongodb does not need to have a strong defined mapped column types because they are not used in schema sync.

I was just using `int` because all other fields were set to `int`, and they are not all `int` columns.

As for SQLite, according to my understanding of the documentation [here](https://www.sqlite.org/datatype3.html), specifying `char` will result with a column of type `text`, so I just used text.

> The following table shows how many common datatype names from more traditional SQL implementations are converted into affinities by the five rules of the previous section. This table shows only a small subset of the datatype names that SQLite will accept. Note that numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions (other than the large global SQLITE_MAX_LENGTH limit) on the length of strings, BLOBs or numeric values.
>
> Example Typenames From TheCREATE TABLE Statement or CAST Expression:  CHARACTER(20), VARCHAR(255), VARYING CHARACTER(255), NCHAR(55), NATIVE, CHARACTER(70), NVARCHAR(100), TEXT, CLOB
> 
> Resulting Affinity: TEXT